### PR TITLE
VACMS 18427 add services query to get facilities api v2

### PR DESCRIPTION
**Pull request by eselkin**
## Summary

- *This work is behind a feature toggle (flipper): NO*
- Add Service model and updated processing for GET /facilities_api/v2/va/id to include LH new services endpoint call
- Sitewide team

## Related issue(s)

- https://github.com/department-of-veterans-affairs/va.gov-cms/issues/18427

## Testing done

- [x] *New code is covered by unit tests*
- Lighthouse removed the access object which contained wait times from the facility query in v1 and FacilitiesApi::V2 removed that because it was no longer in the endpoint. Now we blend two queries to LH in one in the GET /facilities_api/v2/va/id endpoint
- Query /facilities_api/v2/va/vha_640 or any VAMC id and see that in the `data.attributes` object there is a new field access with keys `health` and `effectiveDate`. If the facility has services then `health` will have values and so will `effectiveDate`.

## Screenshots
_Note: Optional_
None

## What areas of the site does it impact?
FacilitiesApi::V2 module GET with an id in the URL path is updated

## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [x]  Events are being sent to the appropriate logging solution
- [x]  Documentation has been updated (link to documentation)
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x]  Feature/bug has a monitor built into Datadog (if applicable)
- [x]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [x]  I added a screenshot of the developed feature

## Requested Feedback

do a curl to vets-api running locally (if you have a lighthouse sandbox key in settings.local.yml) to see that access object is present

`curl  http://localhost:3000/facilities_api/v2/va/vha_648A4`
or
`curl  http://localhost:3000/facilities_api/v2/va/vha_648A4 | jq`

### CODE REVIEW

**Review by Khoa-V-Nguyen — COMMENTED** [review]
Looks good for the `ask_va_api` `health_facilities_spec`. Thank you for updating it.

File: modules/facilities_api/app/services/facilities_api/v2/lighthouse/response.rb [hunk 1]
```diff
@@ -40,6 +40,26 @@ def facility
           V2::Lighthouse::Facility.new(data)
         end
 
+        # services is a string here
+        def facility_with_services(services)
```
Comment (Khoa-V-Nguyen) [inline modules/facilities_api/app/services/facilities_api/v2/lighthouse/response.rb:44]: what do you think about this?  I feel that it's a little bit more readable:

```
        def facility_with_services(services)
          facility = V2::Lighthouse::Facility.new(data)
          parsed_services = parse_services(services)
          health_services = extract_health_services(parsed_services)
          facility.access = build_access_data(health_services)
          facility
        end

        private

        def parse_services(services)
          JSON.parse(services).fetch('data', [])
        end

        def extract_health_services(services_data)
          health_services = services_data.map do |service|
            V2::Lighthouse::Service.new(service)
          end

          health_services.select { |service| service['new'] || service['established'] }
        end

        def build_access_data(health_services)
          if health_services.empty?
            { 'health' => [], 'effectiveDate' => '' }
          else
            { 'health' => health_services, 'effectiveDate' => health_services.first['effectiveDate'] }
          end
        end

```

**Review by Khoa-V-Nguyen — APPROVED** [review]
It looks good. I just have a suggestion but it's not a blocker.

File: config/settings.yml [hunk 2]
```diff
@@ -67,8 +67,8 @@ session_cookie:
   secure: false
 
 sign_in:
-  jwt_encode_key: 'spec/fixtures/sign_in/privatekey.pem'
-  jwt_old_encode_key: 'spec/fixtures/sign_in/privatekey_old.pem'
+  jwt_encode_key: "spec/fixtures/sign_in/privatekey.pem"
```
[the remark this answers is not in the document]
Comment (eselkin) [reply config/settings.yml:70]: No. It must have auto corrected. Will return to its original.

File: modules/facilities_api/app/services/facilities_api/v2/lighthouse/response.rb (hunk 1 above)
Comment (eselkin) [reply modules/facilities_api/app/services/facilities_api/v2/lighthouse/response.rb:44]: Definitely more readable.

Comment (eselkin) [reply modules/facilities_api/app/services/facilities_api/v2/lighthouse/response.rb:44]: Accepted changes. Tests pass.

**Review by Khoa-V-Nguyen — APPROVED** [review]
looks great!

**Review by RachalCassity — APPROVED** [review]

**Pull request merged**